runtime/src/accounts.rs

```diff
@@ -239,9 +239,9 @@ impl Accounts {
             let mut account_deps = Vec::with_capacity(message.account_keys_len());
             let mut rent_debits = RentDebits::default();
             let rent_for_sysvars = feature_set.is_active(&feature_set::rent_for_sysvars::id());
-            let demote_program_write_locks =
-                feature_set.is_active(&feature_set::demote_program_write_locks::id());
-            let is_upgradeable_loader_present = is_upgradeable_loader_present(message);
+            let demote_program_write_lock_features = feature_set
+                .is_active(&feature_set::demote_program_write_locks::id())
+                && feature_set.is_active(&feature_set::restore_write_lock_when_upgradeable::id());
```

Comment:
won't this cause forking if some nodes have updated to have this check and others don't?

Reply:
It could, if I screwed up and activated `demote_program_write_locks` before this lands and is adopted by testnet/devnet. However, my intent is to activate that feature together with `restore_write_lock_when_upgradeable`.
Please let me know if I've missed something, though.

Reply:
Ah got it, that makes sense. It feels to me that we don't need the additional feature switch then. Could we instead replace the old feature switch with this new one? Or just update the behavior of the old switch before enabling it? This approach is doable for sure though

Reply:
Oh yeah, duh, actually we could use only one feature. I did it this way to keep the features in line with what is in v1.6, but there's actually no reason to preserve 2 features there either. I'll clean up v1.6, and then update this to match. Thanks!